src/crystal/system.cr

```diff
@@ -14,4 +14,9 @@ require "./system/unix/hostname"
   require "./system/unix/sysctl_cpucount"
```

Comment:
This whole macro condition should probably be changed to the one posted here: https://github.com/crystal-lang/crystal/pull/6972#issuecomment-445560027.
There's still one UNIX require here: [`system.cr:11`](https://github.com/crystal-lang/crystal/blob/ba46dd965f5b4871ff87f0fb2d936647e052c889/src/crystal/system.cr#L11) which is not in any macro branch.

Reply:
I've changed that to the code from the comment.

Reply:
No, this macro condition is still different from the one in https://github.com/crystal-lang/crystal/pull/6972#issuecomment-445560027.
The UNIX require in [`system.cr:11`](https://github.com/crystal-lang/crystal/blob/ba46dd965f5b4871ff87f0fb2d936647e052c889/src/crystal/system.cr#L11) is still on the top-level but it should be moved to the `flag?(:unix)` macro branch.
The `flag?(:freebsd) || flag?(:openbsd)` macro branch should be removed and this:
```cr
  {% if flag?(:freebsd) || flag?(:openbsd) %}
    require "./system/unix/sysctl_cpucount"
  {% else %}
    require "./system/unix/sysconf_cpucount"
  {% end %}
```
should be added to the `flag?(:unix)` macro branch.